usr.sbin/pkg/config.c

```diff
@@ -66,7 +66,11 @@ static struct config_entry c[] = {
 	[PACKAGESITE] = {
 		PKG_CONFIG_STRING,
 		"PACKAGESITE",
+#ifdef __CheriBSD_version
```

Comment:
Don't these all need to be based on whether `MK_CONFIG` is set? `__CheriBSD_version` is unconditionally defined in CheriBSD.

Reply:
@jrtc27 My initial intention was to use `MK_CHERIBSD` for differences in Makefile code and `__CheriBSD_version` to identify an ABI version but also for differences in C code. We could replace `__CheriBSD_version` checks with `MK_CHERIBSD` checks in C code or define `__CheriBSD_version` only if `MK_CHERIBSD` is set. I'd be leaning towards the latter. What do you think?

Reply:
The do the latter I think we'd have to have `__CheriBSD_version` defined based on some sort of `__CheriBSD__` or `CHERIBSD` we define in `CFLAGS`. When discussing with @kwitaszczyk my though was that `#ifdef __CheriBSD_version` was a slightly more documentary `#if 1`, but I could be persuaded that should add some sort of value to default `CFLAGS` when `MK_CHERIBSD=="yes"`.

Reply:
Well `__CheriBSD_version` is unconditionally defined, and you can't check MK_CHERIBSD in sys/param.h because it's used outside of base. So either `__CheriBSD_version` stays unconditionally defined and you check MK_CHERIBSD for everything in base rather than it, or you have the fun of generating a header (I'd suggest having sys/param.h include a generated header solely for the variable rather than making sys/param.h a template) based on MK_CHERIBSD. But having a MK_CHERIBSD that does basically nothing while tools check for the always-present `__CheriBSD_version` makes no sense to me, might as well drop the option.

Reply:
@brooksdavis @jrtc27 I added `__CheriBSD__` to `CFLAGS` and `CXXFLAGS`. Also, the differences in usr.sbin/pkg/config.c depend on `__CheriBSD__` instead of `__CheriBSD_version` now. What do you think about this approach?

Reply:
This seems fine. You might also want to set `__CheriBSD__` in the kernel as I suspect we'll want to embed the version somewhere eventually.

Reply:
> This seems fine. You might also want to set `__CheriBSD__` in the kernel as I suspect we'll want to embed the version somewhere eventually.

Done. I hope this is what you meant.